app/dot/cpp/login/service/UserService.java

```diff
@@ -51,44 +52,48 @@ public String generateUserWithInvitation(String email, UserRole userRole) {
   }
 
   public String generateResetPasswordUuid(String email) throws UserException {
-    final User user = findByField("email", email);
 
-    if (user == null) {
-      throw new UserException(Error.USER_EMAIL_NOT_FOUND);
-    }
+    try {
+      final User user = findByField("email", email);
+      if (!user.isActive()) {
+        throw new UserException(Error.ACCOUNT_INACTIVE);
+      }
 
-    if (!user.isActive()) {
-      throw new UserException(Error.ACCOUNT_INACTIVE);
-    }
+      final String resetPasswordUuid = UUID.randomUUID().toString();
+      user.setResetPasswordUuid(resetPasswordUuid);
+      save(user);
 
-    final String resetPasswordUuid = UUID.randomUUID().toString();
-    user.setResetPasswordUuid(resetPasswordUuid);
-    save(user);
-
-    logger.debug("{}", user);
-    return resetPasswordUuid;
+      logger.debug("{}", user);
+      return resetPasswordUuid;
+    } catch (EntityNotFoundException e) {
+      throw new UserException(Error.USER_EMAIL_NOT_FOUND);
+    }
   }
 
   public User resetPassword(ResetPasswordRequest resetPasswordRequest, String resetPasswordUuid)
       throws UserException {
     logger.debug("{}", resetPasswordRequest);
     logger.debug("{}", resetPasswordUuid);
 
-    final var user = findByField("resetPasswordUuid", resetPasswordUuid);
+    try {
+      final var user = findByField("resetPasswordUuid", resetPasswordUuid);
 
-    if (user == null) {
-      throw new UserException(Error.NOT_FOUND);
-    }
+      if (user == null) {
+        throw new UserException(Error.NOT_FOUND);
+      }
 
-    final Hash hashedPassword = getHashedPassword(resetPasswordRequest.getPassword());
+      final Hash hashedPassword = getHashedPassword(resetPasswordRequest.getPassword());
 
-    logger.debug("{}", hashedPassword);
-    user.setPassword(hashedPassword.getResult());
-    user.setResetPasswordUuid("");
+      logger.debug("{}", hashedPassword);
+      user.setPassword(hashedPassword.getResult());
+      user.setResetPasswordUuid("");
 
-    logger.debug("{}", user);
-    save(user);
-    return user;
+      logger.debug("{}", user);
+      save(user);
+      return user;
+    } catch (EntityNotFoundException e) {
+      throw new UserException(Error.RESET_PASS_UUID_NOT_FOUND);
```

Comment:
Do we really need a specific error? The generic NOT_FOUND should be enough, we'll have the method name in the application logs to trace it.